org.eclipse.jdt.ls.debug/src/org/eclipse/jdt/ls/debug/adapter/DefaultVariableProvider.java

```diff
@@ -108,11 +114,13 @@ public String valueToString(Value value, Map<String, Object> options) {
     @Override
     public void registerValueFormatter(IValueFormatter formatter, int priority) {
         valueFormatterMap.put(formatter, priority);
+        defaultOptions = null;
```

Comment:
Suggest not using a cache. We don't need to maintain the status when a cache is not used.